### PR TITLE
Connection error handling

### LINKED ISSUE #17: Add error handling when incorrect host/port is entered
Error results in panicked thread

### PR DESCRIPTION
**Pull request by rooneyshuman**
Fixes #17. Re-prompting clients if connection was unsuccessful or if invalid input was entered for port (non-numerical)

### CODE REVIEW

File: src/client_input_handler.rs [hunk 1]
```diff
@@ -13,17 +13,27 @@ pub fn initial_screen() -> String {
...
-    let port_int = port
-        .trim()
-        .parse::<u32>()
-        .expect("could not make port into an int");
+    loop {
+        let mut port = String::new();
+        print!("Enter a port: ");
+        stdout.flush().expect("Error flushing buffer");
+        stdin.read_line(&mut port).expect("Error reading in");
+        match port.trim().parse() {
```
Comment (loganballard) [inline src/client_input_handler.rs]: nice, should we add a check that host is not null?

```
pub fn initial_screen() -> String {
    let stdin = io::stdin();
    let mut stdout = io::stdout();
    let mut host = String::new();
    let port_int: u32;

    loop {
        print!("Enter a host: ");
        stdout.flush().expect("Error flushing buffer");
        stdin.read_line(&mut host).expect("Error reading in");
        if host.trim().is_empty() {
            println!("Enter a valid host!");
            continue;
        }
        break;
    }
    loop {
        let mut port = String::new();
        print!("Enter a port: ");
        stdout.flush().expect("Error flushing buffer");
        stdin.read_line(&mut port).expect("Error reading in");
        match port.trim().parse() {
            Ok(x) => {
                port_int = x;
                break;
            }
            Err(e) => {
                println!("could not make port into an int: {}!", e);
                continue;
            }
        }
    }
    let trimmed_host = host.trim().to_string();
    trimmed_host + &":".to_string() + &port_int.to_string()
}
```

Comment (loganballard) [reply src/client_input_handler.rs]: or possibly refactored into just `get_host_from_input` and `get_port_from_input` helper functions

File: src/client_input_handler.rs [hunk 2]
```diff
@@ -126,16 +136,29 @@ fn list_active_users() -> Msg {
...
-    print!("Which Game Id do you want to join: ");
-    stdout.flush().expect("Client input something nonsensical");
-    stdin.read_line(&mut game_id).expect("I/O error");
+    let game_id: usize;
+    loop {
+        let mut game_id_input = String::new();
+        print!("Which Game ID do you want to join: ");
+        stdout.flush().expect("Client input something nonsensical");
+        stdin.read_line(&mut game_id_input).expect("I/O error");
+        match game_id_input.trim().parse() {
```
Comment (loganballard) [inline src/client_input_handler.rs]: nice

File: src/client_input_handler.rs (hunk 1 above)
Comment (rooneyshuman) [reply src/client_input_handler.rs]: Should definitely add a check for host gp

**Pull request merged**